addons/sale_timesheet/models/project_task.py

```diff
@@ -86,11 +86,18 @@ def _get_last_sol_of_customer(self):
         if not self.partner_id.commercial_partner_id or not self.allow_billable:
             return False
         SaleOrderLine = self.env['sale.order.line']
+        commercial_partner = self.partner_id.commercial_partner_id
+        if hasattr(commercial_partner, '_origin') and commercial_partner._origin:
+            commercial_partner_id = commercial_partner._origin.id
+        else:
+            commercial_partner_id = commercial_partner.id
+        if not isinstance(commercial_partner_id, int):
+            return False
```

Comment:
@abla001 @st-yes 
Actually, I think the solution is wrong, `origin` will contain the data in db and so the user could change the data in the view. we will never use them and always take the data stored in db.

So, if the task is already created and the user alter the task to change the partner, we will still search a SOL for the previous partner instead of the new one. 😕 

Also, I don't instance why we need to do that:
```py
        if not isinstance(commercial_partner_id, int):
            return False
```
to me we could just check if the recordset is empty in the both case (in self and in origin) and it is good enough. 🤔